packages/contracts/contracts/plugin/PluginManager.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier:    MIT
+
+pragma solidity 0.8.10;
+
+import "../utils/PluginERC1967Proxy.sol";
+import "../core/permission/BulkPermissionsLib.sol";
+import "./PluginConstants.sol";
+
+/// @notice Abstract Plugin Factory that dev's have to inherit from for their factories.
+abstract contract PluginManager is PluginConstants {
+    bytes4 public constant PLUGIN_FACTORY_INTERFACE_ID = type(PluginManager).interfaceId;
+
+    // TODO: come up with a better naming for isWhereAddress, isWhoAddress, or come up with another struct.
+    struct RequestedPermission {
+        BulkPermissionsLib.Operation op;
+        uint256 where; // index from relatedContracts or the actual address
+        bool isWhereAddress; // whether or not `where` is index from relatedContracts or address directly.
+        uint256 who; // index from relatedContracts or the actual address
+        bool isWhoAddress; // whether or not `who` is index from relatedContracts or address directly.
+        address oracle;
+        bytes32 permissionId;
+    }
+
+    /// @notice creates Permission struct 
+    /// @param op Whether grants, revokes, freezes...
+    /// @param where index from the dev's deployed addresses array where permission will be set.
+    /// @param who index from the dev's deployed addresses array
+    /// @param permissionId permissionId that will be set
+    /// @return Permission The final permission struct
+    function createPermission(
```

Comment:
I would rename the 4 of them `buildPermission()`. Create sounds like you are registering a permission somewhere, but here you're just building an object and returning it.